### PR TITLE
Woo: 2 of 3 - Sectioned order list and status filtering

### PR DESCRIPTION
**Pull request by AmandaRiu**
*** NOT READY FOR REVIEW ***
TODO
- [ ] ~Fix localization issue in SQL query~ (opened separate ticket: https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/1273)
- [ ] ~Add logic to delete all the `WCOrderSummaryModel` records when `fetchFirstPage()` is called.~ (opened separate ticket: https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/1272)
- [x] Create an ApiResponse class for `WCOrderSummaryModel`

This is part 2 of the hack week project to add `ListStore` management for Woo Order lists. PR #1235  must first be merged before this one.

This PR adds sectioned capabilities to the order list as well as filtering by order status.

Adds the following:
- Introduces `WCOrderSummaryModel` for sectioned list capabilities
- Adds basic section headers and comparison in DiffCallback
- Adds support for order list filtering, also added to the example app order list test page.
- Uses SQL to determine what `TimeGroup` each order belongs in. Since this calculation has the potential to change whenever the date changes, the value is not stored in the DB. 
- Adds unit tests for WCOrderListDescriptor
- Adds additional connected tests

Does not include (handled in the final PR):
- Fetching orders that have been modified since last fetch

### CODE REVIEW

**Comment by AmandaRiu** [comment]
@malinajirka This PR is finally ready for your review. I handled all the comments you left in the first PR (#1235) in this PR as well. I moved a couple of the line items in the original description into separate tickets so they won't hold up this PR and this can keep chugging along.

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/OrderSqlUtils.kt [hunk 1]
```diff
@@ -1,22 +1,56 @@
...
+        WellSql.insert(orderSummaries).asSingleTransaction(true).execute()
+    }
+
+    // TODO verify and correct localization or figure out why sometimes the time may be off
+    fun getOrderSummariesForRemoteIds(site: SiteModel, remoteOrderIds: List<RemoteId>): List<WCOrderSummaryModel> {
+        if (remoteOrderIds.isEmpty()) {
+            return emptyList()
+        }
+        val readableDb = WellSql.giveMeReadableDb()
+        val sql = "SELECT *,\n" +
```
[the remark this answers is not in the document]
Comment (AmandaRiu) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/OrderSqlUtils.kt]: I thought about this and while I believe the SQL is faster, it's also true that it's much more difficult to test and troubleshoot. It already had some localization issues present. So taking your review into consideration, I decided to go back to dealing with this directly in code in d0b92ceab8aee19de4e2148d40385d49dad20ade

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/OrderSqlUtils.kt [hunk 2]
```diff
@@ -243,4 +281,13 @@ object OrderSqlUtils {
                     .endWhere()
                     .orderBy(WCOrderShipmentProviderModelTable.COUNTRY, SelectQuery.ORDER_ASCENDING)
                     .asModel
+
+    private fun getOrderSummaryModelFromCursor(cursor: Cursor): WCOrderSummaryModel {
+        return WCOrderSummaryModel().apply {
+            localSiteId = cursor.getInt(cursor.getColumnIndex(WCOrderSummaryModelTable.LOCAL_SITE_ID))
+            remoteOrderId = cursor.getLong(cursor.getColumnIndex(WCOrderSummaryModelTable.REMOTE_ORDER_ID))
+            dateCreated = cursor.getString(cursor.getColumnIndex(WCOrderSummaryModelTable.DATE_CREATED))
+            timeGroup = TimeGroup.valueOf(cursor.getString(cursor.getColumnIndex("TIME_GROUP")))
```
[the remark this answers is not in the document]
Comment (AmandaRiu) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/OrderSqlUtils.kt]: I agree! But it doesn't matter now :) I've removed this altogether in d0b92ceab8aee19de4e2148d40385d49dad20ade

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/OrderSummaryApiResponse.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package org.wordpress.android.fluxc.network.rest.wpcom.wc.order
+
+import org.wordpress.android.fluxc.network.Response
+
+@Suppress("PropertyName")
```
[the remark this answers is not in the document]
Comment (AmandaRiu) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/OrderSummaryApiResponse.kt]: Totally agree and I've implemented this change. The commits are a bit of a mess - I apologize for that. Had a refactor mishap. Starts at commit 9aa2fc26f8ec805d6b9644f465311bd22aa4f179

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/OrderRestClient.kt [hunk 4]
```diff
@@ -592,6 +622,12 @@ class OrderRestClient(
         }
     }
 
+    private fun orderDateCreatedFromOrderResponse(response: OrderApiResponse): String =
+            response.date_created_gmt?.let { "${it}Z" } ?: "" // Store the date in UTC format
```
[the remark this answers is not in the document]
Comment (AmandaRiu) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/OrderRestClient.kt]: Done in 1a087dcf3b528d50cd0c790ee6a2f981a5f15b80 👍

**Comment by AmandaRiu** [comment]
@malinajirka Thank you for the review! Ready for a second round!

**Comment by AmandaRiu** [comment]
Updated branch with the AndroidX changes

**Comment by malinajirka** [comment]
LGTM, great job!!! Thanks;)

**Pull request merged**